docs/StardustDocs/topics/dataSources/Parquet.md

```diff
@@ -0,0 +1,192 @@
+
+### Batch Size Tuning
+
+- **Default**: (typically 1024) `ARROW_PARQUET_DEFAULT_BATCH_SIZE`
+- **Small files** (< 100MB): Use default
+- **Large files** (> 1GB): Increase to `64 * 1024` or `128 * 1024`
+- **Memory constrained**: Decrease to `256` or `512`
+
+```kotlin
+// For large files with enough memory
+DataFrame.readParquet("large_file.parquet", batchSize = 64L * 1024)
+
+// For memory-constrained environments  
+DataFrame.readParquet("file.parquet", batchSize = 256L)
+```
+
+### Nullability Inference
+
+Controls how nullable columns are handled:
+
+```kotlin
+// Infer nullability from data (default)
+DataFrame.readParquet("file.parquet", nullability = NullabilityOptions.Infer)
+
+// Treat all columns as nullable
+DataFrame.readParquet("file.parquet", nullability = NullabilityOptions.Enable)
+
+// Treat all columns as non-null (may cause runtime errors)
+DataFrame.readParquet("file.parquet", nullability = NullabilityOptions.Disable)
+```
+
+## About Parquet
+
+[Apache Parquet](https://parquet.apache.org/) is an open-source, column-oriented data file format designed for efficient data storage and retrieval. It provides several advantages:
+
+- **Columnar storage**: Data is stored column-by-column, which enables efficient compression and encoding schemes
+- **Schema evolution**: Supports adding new columns without breaking existing data readers
+- **Efficient querying**: Optimized for analytics workloads where you typically read a subset of columns
+- **Cross-platform**: Works across different programming languages and data processing frameworks
+- **Compression**: Built-in support for various compression algorithms (GZIP, Snappy, etc.)
+
+Parquet files are commonly used in data lakes, data warehouses, and big data processing pipelines. They're frequently created by tools like Apache Spark, Pandas, Dask, and various cloud data services.
+
+## Typical use cases
+
+- Exchanging columnar datasets between Spark and Kotlin/JVM applications.
+- Analytical workloads where columnar compression and predicate pushdown matter.
+- Reading data exported from data lakes and lakehouse tables (e.g., from Spark, Hive, or Delta/Iceberg exports).
+
+If you want to see a complete, realistic data‑engineering example using Spark and Parquet with Kotlin DataFrame,
+check out the [example project](https://github.com/Kotlin/dataframe/tree/master/examples/idea-examples/spark-parquet-dataframe).
+
+### Performance tips
+
+- **Column selection**: Because the ` readParquet ` method reads all columns, use DataFrame operations like `select()` immediately after reading to reduce memory usage in later operations
+- **Predicate pushdown**: Currently not supported—filtering happens after data is loaded into memory
+- Use Arrow‑compatible JVMs as documented in
+  [Apache Arrow Java compatibility](https://arrow.apache.org/docs/java/install.html#java-compatibility).
+- Adjust `batchSize` if you read huge files and need to tune throughput vs. memory.
+
+## Limitations
```

Comment:
this info appears twice on the page. Seems a bit overkill. I think once is enough :)